### PR TITLE
Themes: Enable item removal in ThemeQueryManager

### PR DESCRIPTION
**Pull request by seear**
No visual differences. Includes unit tests.

Implements a `removeItems()` method for `ThemeQueryManager` that doesn't rely on the parent `receive` method, which has already been overridden.

At a later date we should re-implement `ThemeQueryManager` without relying on a parent class at all, since we are only benefitting from a couple of getters.

### CODE REVIEW

File: client/lib/query-manager/theme/index.js [hunk 1]
```diff
@@ -123,6 +124,29 @@ export default class ThemeQueryManager extends PaginatedQueryManager {
...
+	 * This method is overridden to call the parent receive() method, which
+	 * maintains the ability to delete items, unlike the cut-down receive()
+	 * method in this class.
+	 *
+	 * @param  {String[]}     itemKeys Keys of items to remove
+	 * @return {QueryManager}          New instance if changed, or same
+	 *                                 instance otherwise
+	 */
+	removeItems( itemKeys = [] ) {
+		return PaginatedQueryManager.prototype.receive.call( this,
```
Comment (ockham) [inline client/lib/query-manager/theme/index.js]: Line can be shortened to

```js
return super.receive(
```

OOP FTW 😁

File: client/lib/query-manager/theme/test/index.js [hunk 2]
```diff
@@ -1,17 +1,57 @@
...
-	// TODO
+	let manager;
+
+	beforeEach( () => {
+		manager = new ThemeQueryManager();
+	} );
+
+	describe( '#removeItem()', () => {
+		it( 'should remove an item by its item key', () => {
+			manager = manager.receive( [ { ID: 144 } ] );
```
Comment (ockham) [inline client/lib/query-manager/theme/test/index.js]: This line could go to a `beforeEach` inside the `describe` block, I think.

**Comment by ockham** [comment]
The one thing I'm not entirely sure about is how this is going to affect ordering of items in a query. This was the main reason why I had to override `receive` method for `ThemeQueryManager` in the first place (and wasn't even able to use the parent class' in my implementation). I think that `receive` will always iterate over its existing items and re-order them by the sort order specified by the `matches` method (which we aren't going to implement for themes for a good reason).

What about iterating over queries and `omit`ting the deleted item ourselves? Since we're not relying on the parent's `receive` elsewhere, this might be preferable (as it's probably more predictable)...

**Comment by seear** [comment]
> What about iterating over queries and omitting the deleted item ourselves?

Strikes me now that this may be cleaner and more understandable anyway. I'll look into it.

File: client/lib/query-manager/theme/test/index.js (hunk 2 above)
Comment (seear) [inline client/lib/query-manager/theme/test/index.js]: These tests are taken from [`QueryManager`](https://github.com/Automattic/wp-calypso/blob/24f141b6c19ddc699ace7e81f3d08dc02d99e147/client/lib/query-manager/test/index.js#L150) so we may want to keep them the same.

**Comment by seear** [comment]
Closing in favour of #10709

**Pull request closed without merging**